### PR TITLE
Rename `ffi/diplomat` to `ffi/capi`

### PR DESCRIPTION
**Pull request by robertbastian**
The crate is called `icu_capi`, and it's confusing to have two crates in `diplomat` directories in my workspace.

### CODE REVIEW

**Review by sffc — APPROVED** [review]
LGTM but wait for @Manishearth approval too

**Review by Manishearth — DISMISSED** [review]
Yeah I've found this annoying too

**Review by Manishearth — APPROVED** [review]

**Pull request merged**
